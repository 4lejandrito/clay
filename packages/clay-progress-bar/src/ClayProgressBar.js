import 'clay-icon';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayProgressBar.soy.js';

/**
 * Metal Clay Button component.
 * @extends Component
 */
class ClayProgressBar extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayProgressBar.STATE = {
	/**
	 * CSS classes to be applied to the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayProgressBar
	 * @type {?(string|undefined)}
	 */
	elementClasses: Config.string(),

	/**
	 * Id to be applied to the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayProgressBar
	 * @type {?(string|undefined)}
	 */
	id: Config.string(),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @default undefined
	 * @instance
	 * @memberof ClayProgressBar
	 * @type {!string}
	 */
	spritemap: Config.string(),

	/**
	 * Progress bar state. Completed state can be defined here but also is
	 * automatically set if value is equal to maxValue.
	 * @default undefined
	 * @instance
	 * @memberof ClayProgressBar
	 * @type {?string}
	 */
	status: Config.oneOf(['complete', 'warning']),

	/**
	 * The progress value of the bar.
	 * @default undefined
	 * @instance
	 * @memberof ClayProgressBar
	 * @type {?number}
	 */
	value: {
		validator: value => value >= 0 && value <= 100,
		value: 0,
	},
};

defineWebComponent('clay-progress-bar', ClayProgressBar);

Soy.register(ClayProgressBar, templates);

export {ClayProgressBar};
export default ClayProgressBar;
