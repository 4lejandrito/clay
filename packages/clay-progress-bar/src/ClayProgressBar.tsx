/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import classNames from 'classnames';

interface Props extends React.HTMLAttributes<HTMLDivElement> {
	feedback?: boolean;
	status?: 'success' | 'warning';
	value: number;
}

const ClayProgressBar: React.FunctionComponent<Props> = ({
	children,
	className,
	feedback,
	status,
	value,
	...otherProps
}) => {
	let addon = children;

	if (feedback) {
		addon = <div className="progress-group-feedback">{addon}</div>;
	}

	return (
		<div
			{...otherProps}
			className={classNames(className, 'progress-group', {
				[`progress-${status}`]: status,
			})}
		>
			<div className="progress">
				<div
					aria-valuemax={100}
					aria-valuemin={0}
					aria-valuenow={value}
					className="progress-bar"
					role="progressbar"
					style={{width: `${value}%`}}
				/>
			</div>

			{addon && <div className="progress-group-addon">{addon}</div>}
		</div>
	);
};

export default ClayProgressBar;
