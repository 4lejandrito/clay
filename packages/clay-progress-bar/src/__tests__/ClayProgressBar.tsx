/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayProgressBar from '../ClayProgressBar';

describe('ClayProgressBar', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(
			<ClayProgressBar />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
