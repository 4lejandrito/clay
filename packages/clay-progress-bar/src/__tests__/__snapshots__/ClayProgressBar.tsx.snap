// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayProgressBar renders 1`] = `
<div
  className="progress-group"
>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={50}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
</div>
`;

exports[`ClayProgressBar renders as a different status 1`] = `
<div
  className="progress-group progress-warning"
>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={50}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    className="progress-group-addon"
  >
    50%
  </div>
</div>
`;

exports[`ClayProgressBar renders with additional content 1`] = `
<div
  className="progress-group"
>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={50}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    className="progress-group-addon"
  >
    50%
  </div>
</div>
`;

exports[`ClayProgressBar wraps content in feedback classname 1`] = `
<div
  className="progress-group"
>
  <div
    className="progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={50}
      className="progress-bar"
      role="progressbar"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    className="progress-group-addon"
  >
    <div
      className="progress-group-feedback"
    >
      50%
    </div>
  </div>
</div>
`;
